newspaper/extractors.py

```diff
@@ -164,6 +164,20 @@ def parse_byline(search_str):
 
     def get_title(self, doc):
         """Fetch the article title and analyze it
+
+        Assumptions:
+        - title tag is the most reliable (inherited from Goose)
+        - h1, if properly detected, is the best (visible to users)
+        - og:title and h1 can help improve the title extraction
+        - python == is too strict, often we need to compare fitlered
+          versions, i.e. lowercase and ignoring special chars
+
+        Explicit rules:
+        1. title == h1, no need to split
+        2. h1 similar to og:title, use h1
+        3. title contains h1, title contains og:title, len(h1) > len(og:title), use h1
+        4. title starts with og:title, use og:title
+        5. title starts with h1 and h1 long enough, use h1
```

Comment:
I think that this rule (5) can cause lots of wrong results and a fallback to title is preferable here.